src/components/gds-heading/_index.scss

```diff
@@ -1,6 +1,8 @@
+@use "../../styles/typography" as *;
+
 @mixin base {
   display: block;
-  font-family: Gilroy, Arial, Helvetica, sans-serif;
+  font-family: $label-font-family;
```

Comment:
So why not `font-family: var(--label-font-family);` 

and in typography:

```scss
:root {
  --heading-font-family: Gilroy, Arial, Helvetica, sans-serif;
  --body-font-family: Gilroy, Arial, Helvetica, sans-serif;
  --label-font-family: Gilroy, Arial, Helvetica, sans-serif;

  --button-font-family: var(--body-font-family);
}
```

If we do this we cannot `@use` the typography file though, or it would override any global variables in a project using the library. We'd just have to rely on someone having included it before. Alternatively and maybe better is to tell projects to override variables with:

```scss
:root:root {
  --heading-font-family: serif;
}
```

Not sure if that's ideal, could actually be okay? Another but a bit verbose way is to check if the variables exist before redefining them.

```scss
:root {
  --heading-font-family: var(--heading-font-family, Gilroy, Arial, Helvetica, sans-serif);
  --body-font-family: var(--body-font-family, Gilroy, Arial, Helvetica, sans-serif);
  --label-font-family: var(--label-font-family, Gilroy, Arial, Helvetica, sans-serif);

  --button-font-family: var(--body-font-family);
}
```

Reply:
More on this, what I would actually like is to have things defined within the components.

So that `src/components/gds-heading/_index.scss` looks like:

```scss
@use '../../styles/typography';

@mixin base {
  display: block;
  font-size: var(--heading-font-size);
  line-height: var(--heading-line-height);
  font-family: var(--heading-font-family);
  font-weight: var(--heading-font-weight, 600);
  letter-spacing: var(--heading-letter-spacing, 0);
}

// H1
@mixin heading-xxl {
  --heading-font-size: 54px;
  --heading-line-height: 52px;

  @include desktop {
    --heading-font-size: 72px;
    --heading-line-height: 74px;
  }

  @include base;
}
```

`src/components/gds-heading/gds-heading.scss` looks like:

```scss
.gds-heading-xxl {
  @include heading-xxl;
}
```

And `src/styles/typography.scss` looks like:

```scss
:root {
  --heading-font-family: Gilroy, Arial, Helvetica, sans-serif;
}
```

Reply:
Maybe go with what you have for now and then we play around and experiment how we can override things later when we actually have generogrowth using the gds components?

Reply:
Thanks for thinking about this! 

Yeah, we definitely need to use custom variables for theming. So how about this?

In `styles/typography.scss´
```scss
@mixin base {
  --heading-font-family: Gilroy, Arial, Helvetica, sans-serif;
}
:root {
  
}
```

In `styles/_index.scss´
```scss
@use "./typography" as *;

:root {
  @include typography;
}
```

Reply:
I had some trouble getting the sass injected to the storybook so I just went with index.css file for custom variables.

Reply:
We can fix this later.